File: backend/app/models/photo.py
from app.extensions import db
from app.models import BaseModel

class Photo(BaseModel):
    id = db.Column(db.Integer, primary_key=True)
    width = db.Column(db.Integer)
    height = db.Column(db.Integer)
    url = db.Column(db.String(255), nullable=False)
    photographer = db.Column(db.String(255))
    photographer_url = db.Column(db.String(255))
    photographer_id = db.Column(db.Integer)
    avg_color = db.Column(db.String(32))
    src_original = db.Column(db.String(255))
    src_large2x = db.Column(db.String(255))
    src_large = db.Column(db.String(255))
    src_medium = db.Column(db.String(255))
    src_small = db.Column(db.String(255))
    src_portrait = db.Column(db.String(255))
    src_landscape = db.Column(db.String(255))
    src_tiny = db.Column(db.String(255))
    alt = db.Column(db.String(255))
    likes = db.relationship("Like", back_populates="photo")

File: app/models/photo.rb
# == Schema Information
#
# Table name: photos
#
#  id               :integer          not null, primary key
#  alt              :text
#  avg_color        :string
#  height           :integer
#  likes_count      :integer          default(0), not null
#  photographer     :string
#  photographer_url :string
#  src_landscape    :string
#  src_large        :string
#  src_large2x      :string
#  src_medium       :string
#  src_original     :string
#  src_portrait     :string
#  src_small        :string
#  src_tiny         :string
#  url              :string
#  width            :integer
#  created_at       :datetime         not null
#  updated_at       :datetime         not null
#  pexels_id        :integer
#  photographer_id  :integer
#
class Photo < ApplicationRecord
  has_many :likes, dependent: :destroy
  has_many :users, through: :likes

  validates :pexels_id, presence: true, uniqueness: true
  validates :photographer, presence: true
  validates :url, presence: true

  scope :ordered_by_date, -> { order(created_at: :desc) }
end
